Adds a switch (that publishes on/off messages and subscribes to a topic to receive on/off from an external source

jacquesdriessen opened the pull request:
Thanks for creating this library by the way, love it. Anyway, need it, if you think it could be useful feel free to include in the main branch or drop if you feel it's not.

andremiller commented:
Thanks @jacquesdriessen, that's great!

With these changes, would others not using this functionality be affected in any way if they update to this version?

I don't have the time right now to do thorough testing, but I don't want the plugin to break for anyone currently using it!

jacquesdriessen commented:
Hold off merging for now. bottomline right now the sensors dont work so i messed up. Will let you know when i have something that i know doesnt break things,

jacquesdriessen commented:
Andrew,
- Wrote the code against the pimatic version of your plugin. That worked
  together / no issues.
- Then pulled down the latest from github, backported, tested code I added,
  which worked, but forgot to test basic functionality.
- Long & short of it is that the latest version from github (without what I
  added) and that doesn't work on my setup. Eg. my testing platform isn't
  sound so I am with you, don't release it please as we can't guarantee it
  won't break anything else (as I simply can't test it).

For myself have installed the pimatic version of your library, created a
separate dir and just separated out the code base. Now as the last I want
to do is create separate forks of what is a great piece of software, but at
the same time it might at some time come in handy should you decide to want
to add such functionality, have attached the files to this mail so when the
time comes it might save you some time.

Thanks for your quick response, the great library (I wouldn't have known
how to start without that) and your patience with a newbie to github and
json.

Jacques.

andremiller commented:
Thanks @jacquesdriessen 

The github and pimatic versions should be the same, because pimatic installs from NPM, and NPM pulls from github.

See if you get your test setup working. Would be nice to merge this without creating a fork!

jacquesdriessen commented:
Andre, will have a look at this early next week (coming week won't have
time unfortunately)

On 28 November 2015 at 22:58, Andre Miller notifications@github.com wrote:

> Thanks @jacquesdriessen https://github.com/jacquesdriessen
> 
> The github and pimatic versions should be the same, because pimatic
> installs from NPM, and NPM pulls from github.
> 
> See if you get your test setup working. Would be nice to merge this
> without creating a fork!
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/andremiller/pimatic-mqtt-simple/pull/9#issuecomment-160338979
> .

wutu commented:
I tested this pulll against ESPEasy and works well.
Pls pull up soon :smile:
But not with this github version (json version). NPM version works well for me with mosquitto broker.

Nice plugin btw. Thx
